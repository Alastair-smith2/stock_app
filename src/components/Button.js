import React, { Component } from "react";
import { View, Text, TouchableHighlight, StyleSheet } from "react-native";

export default class Button extends Component {
  constructor(props) {
    super(props);

    this.onPress = this.onPress.bind(this);
  }
  onPress = () => {
    this.props.buttonPress();
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this.onPress}
        style={[styles.container, this.props.buttonStyle]}
      >
        <Text style={styles.label}>{this.props.title}</Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  label: {
    fontSize: 20,
    color: "#fff"
  }
});
