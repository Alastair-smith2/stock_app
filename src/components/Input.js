import React, { Component } from "react";
import { View, Text, TextInput, StyleSheet } from "react-native";

export default class Input extends Component {
  onTextChange = props => {
    this.props.onTextInput(this.props.type, props);
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.label}>{this.props.label}</Text>
        <TextInput
          value={this.props.value}
          placeholder={`Please enter you ${this.props.type}`}
          onChangeText={text => this.onTextChange(text)}
          type={this.props.type}
          underlineColorAndroid={"transparent"}
          autoCapitalize={"none"}
          style={styles.textInput}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignContent: "center"
  },
  label: {
    fontSize: 16,
    fontFamily: "Futura"
  },
  textInput: {
    flex: 1.5,
    height: 40,
    borderColor: "#BBB",
    borderWidth: 1,
    padding: 10
  }
});
