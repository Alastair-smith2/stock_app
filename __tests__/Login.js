import React from "react";
import { shallow } from "enzyme";
import Button from "../src/components/Button.js";
import Input from "../src/components/Input.js";
import {
  Login,
  mapDispatchToProps,
  mapStateToProps
} from "../src/screens/Login.js";

describe("The Login screen", () => {
  let wrapper;
  let navigation = {
    dispatch: jest.fn()
  };
  let appData = {
    authenticated: false,
    stockData: [],
    stockCode: "",
    currencyData: null,
    dataFetched: false,
    isFetching: false,
    error: false
  };
  let login = jest.fn();

  let initialState = {
    appData: {
      authenticated: false,
      stockData: [],
      stockCode: "",
      currencyData: null,
      dataFetched: false,
      isFetching: false,
      error: false
    }
  };

  beforeEach(() => {
    wrapper = shallow(
      <Login appData={appData} navigation={navigation} login={login} />
    );
    wrapper.setState({ username: "", password: "" });
    navigation.dispatch.mockReset();
    login.mockReset();
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Features", () => {
    it("Should have two Input components", () => {
      expect(wrapper.find(Input)).toHaveLength(2);
    });

    it("Should have a Button component", () => {
      expect(wrapper.find(Button)).toHaveLength(1);
    });

    it("On pressing the Button should dispatch to the next page", () => {
      wrapper
        .find(Button)
        .props()
        .buttonPress();
      expect(login.mock.calls.length).toEqual(0);
      wrapper.setState({ username: "Admin", password: "Pass" });
      wrapper
        .find(Button)
        .props()
        .buttonPress();
      expect(navigation.dispatch.mock.calls.length).toEqual(1);
      expect(login.mock.calls.length).toEqual(1);
    });

    it("Should an initial state that is a pair of empty strings", () => {
      expect(wrapper.state("username")).toEqual("");
      expect(wrapper.state("password")).toEqual("");
    });

    it("Should have a function that handles the textChange", () => {
      wrapper.instance().handleTextChange("username", "Alastair");
      expect(wrapper.state("username")).toEqual("Alastair");
    });
  });

  describe("Redux connect", () => {
    it("should call fetch data action via mapDispatchToProps", () => {
      const dispatchSpy = jest.fn();
      const { login } = mapDispatchToProps(dispatchSpy);
      login();
      expect(dispatchSpy.mock.calls.length).toEqual(1);
    });

    it("Should return the same state as props via mapStateToProps", () => {
      let initialData = {
        appData: 1
      };
      expect(mapStateToProps(initialData).appData).toEqual(initialData.appData);
    });
  });
});
