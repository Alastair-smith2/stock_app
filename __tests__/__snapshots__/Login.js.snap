// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Login screen It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    appData={
      Object {
        "authenticated": false,
        "currencyData": null,
        "dataFetched": false,
        "error": false,
        "isFetching": false,
        "stockCode": "",
        "stockData": Array [],
      }
    }
    login={[MockFunction]}
    navigation={
      Object {
        "dispatch": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.2,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Futura",
                "fontSize": 24,
                "fontWeight": "500",
              }
            }
          >
            Stock & Exchange Search
          </Text>
        </View>,
        <View
          style={
            Object {
              "flex": 0.1,
              "margin": 10,
            }
          }
        >
          <Input
            label="Username"
            onTextInput={[Function]}
            type="username"
            value=""
          />
        </View>,
        <View
          style={
            Object {
              "flex": 0.1,
              "margin": 10,
            }
          }
        >
          <Input
            label="Password"
            onTextInput={[Function]}
            type="password"
            value=""
          />
        </View>,
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.3,
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Button
            buttonPress={[Function]}
            buttonStyle={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EA572D",
                "borderRadius": 60,
                "flex": 0.5,
                "justifyContent": "center",
                "padding": 15,
              }
            }
            title="Submit"
          />
        </View>,
      ],
      "style": Object {
        "backgroundColor": "#fff",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontFamily": "Futura",
                "fontSize": 24,
                "fontWeight": "500",
              }
            }
          >
            Stock & Exchange Search
          </Text>,
          "style": Object {
            "alignItems": "center",
            "flex": 0.2,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Stock & Exchange Search",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontFamily": "Futura",
              "fontSize": 24,
              "fontWeight": "500",
            },
          },
          "ref": null,
          "rendered": "Stock & Exchange Search",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            label="Username"
            onTextInput={[Function]}
            type="username"
            value=""
          />,
          "style": Object {
            "flex": 0.1,
            "margin": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Username",
            "onTextInput": [Function],
            "type": "username",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            label="Password"
            onTextInput={[Function]}
            type="password"
            value=""
          />,
          "style": Object {
            "flex": 0.1,
            "margin": 10,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "label": "Password",
            "onTextInput": [Function],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Button
            buttonPress={[Function]}
            buttonStyle={
              Object {
                "alignItems": "center",
                "backgroundColor": "#EA572D",
                "borderRadius": 60,
                "flex": 0.5,
                "justifyContent": "center",
                "padding": 15,
              }
            }
            title="Submit"
          />,
          "style": Object {
            "alignItems": "center",
            "flex": 0.3,
            "flexDirection": "row",
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonPress": [Function],
            "buttonStyle": Object {
              "alignItems": "center",
              "backgroundColor": "#EA572D",
              "borderRadius": 60,
              "flex": 0.5,
              "justifyContent": "center",
              "padding": 15,
            },
            "title": "Submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.2,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Futura",
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              Stock & Exchange Search
            </Text>
          </View>,
          <View
            style={
              Object {
                "flex": 0.1,
                "margin": 10,
              }
            }
          >
            <Input
              label="Username"
              onTextInput={[Function]}
              type="username"
              value=""
            />
          </View>,
          <View
            style={
              Object {
                "flex": 0.1,
                "margin": 10,
              }
            }
          >
            <Input
              label="Password"
              onTextInput={[Function]}
              type="password"
              value=""
            />
          </View>,
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 0.3,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Button
              buttonPress={[Function]}
              buttonStyle={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EA572D",
                  "borderRadius": 60,
                  "flex": 0.5,
                  "justifyContent": "center",
                  "padding": 15,
                }
              }
              title="Submit"
            />
          </View>,
        ],
        "style": Object {
          "backgroundColor": "#fff",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontFamily": "Futura",
                  "fontSize": 24,
                  "fontWeight": "500",
                }
              }
            >
              Stock & Exchange Search
            </Text>,
            "style": Object {
              "alignItems": "center",
              "flex": 0.2,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Stock & Exchange Search",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontFamily": "Futura",
                "fontSize": 24,
                "fontWeight": "500",
              },
            },
            "ref": null,
            "rendered": "Stock & Exchange Search",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              label="Username"
              onTextInput={[Function]}
              type="username"
              value=""
            />,
            "style": Object {
              "flex": 0.1,
              "margin": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "label": "Username",
              "onTextInput": [Function],
              "type": "username",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              label="Password"
              onTextInput={[Function]}
              type="password"
              value=""
            />,
            "style": Object {
              "flex": 0.1,
              "margin": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "label": "Password",
              "onTextInput": [Function],
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              buttonPress={[Function]}
              buttonStyle={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EA572D",
                  "borderRadius": 60,
                  "flex": 0.5,
                  "justifyContent": "center",
                  "padding": 15,
                }
              }
              title="Submit"
            />,
            "style": Object {
              "alignItems": "center",
              "flex": 0.3,
              "flexDirection": "row",
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "buttonPress": [Function],
              "buttonStyle": Object {
                "alignItems": "center",
                "backgroundColor": "#EA572D",
                "borderRadius": 60,
                "flex": 0.5,
                "justifyContent": "center",
                "padding": 15,
              },
              "title": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
